workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerConnector.cs

```diff
@@ -196,6 +207,67 @@ protected virtual string SelectDeploymentName(DeploymentList deployments)
             return null;
         }
 
+        protected virtual string SelectLoginToken(List<LoginTokenDetails> loginTokens)
+        {
+            return null;
+        }
+
+        protected virtual string RetrievePlayerIdentityToken(string authToken, string playerId, string displayName)
+        {
+            var result = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+                RuntimeConfigDefaults.LocatorHost,
+                RuntimeConfigDefaults.DevelopmentPlayerIdentityTokenPort,
+                new PlayerIdentityTokenRequest
+                {
+                    DevelopmentAuthenticationTokenId = authToken,
+                    PlayerId = playerId,
+                    DisplayName = displayName,
+                }
+            ).Get();
+
+            if (!result.HasValue)
+            {
+                throw new ConnectionFailedException("Did not receive a player identity token.",
+                    ConnectionErrorReason.FailedAuthentication);
+            }
+
+            if (!string.IsNullOrEmpty(result.Value.Error))
+            {
+                throw new ConnectionFailedException(result.Value.Error, ConnectionErrorReason.FailedAuthentication);
+            }
+
+            return result.Value.PlayerIdentityToken;
+        }
+
+        protected virtual List<LoginTokenDetails> RetrieveLoginToken(string workerType, string playerIdentityToken)
+        {
+            var loginTokenRequestResult = DevelopmentAuthentication.CreateDevelopmentLoginTokensAsync(
+                RuntimeConfigDefaults.LocatorHost,
+                444,
+                new LoginTokensRequest
+                {
+                    WorkerType = workerType,
+                    PlayerIdentityToken = playerIdentityToken,
+                    UseInsecureConnection = false,
+                    DurationSeconds = 120,
+                }
+            ).Get();
+
+            if (!loginTokenRequestResult.HasValue)
+            {
+                throw new ConnectionFailedException("Did not receive any login tokens back.",
+                    ConnectionErrorReason.InvalidConfig);
+            }
+
+            if (loginTokenRequestResult.Value.Status != ConnectionStatusCode.Success)
+            {
+                throw new ConnectionFailedException($"Failed to retrieve login token, " +
```

Comment:
I'd rather have a slightly longer line length than the string concatenation here.

Mild preference though